### PR TITLE
Implement categories component

### PR DESCRIPTION
**Pull request by dgutoryan**

### CODE REVIEW

File: src/app/categories/categories.component.html [hunk 1]
```diff
@@ -1,3 +1,17 @@
-<p>
-  categories works!
-</p>
+<div class="categories-list">
+  <div
+    class="category-item"
+    *ngFor="let category of categories$ | async"
+  >
+    <div
+      class="category-img"
```
Comment (AndreyLukashenko) [inline src/app/categories/categories.component.html]: Use full word `category-image`

File: src/app/categories/categories.component.html [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
+      [ngStyle]="{'background-image': 'url(' + (category.image || '') + ')'}"
+    ></div>
+    <span class="category-name">
+      {{category.name}}
+    </span>
+    <span class="category-recipies">
+        {{category.recipies_amount}} {{category.recipies_amount == 1 ? 'recipe' : 'recipes'}}
```
Comment (AndreyLukashenko) [inline src/app/categories/categories.component.html]: Let use camelcase `recipiesAmount`

Comment (AndreyLukashenko) [reply src/app/categories/categories.component.html]: In JavaScript always use strict comparison. `===`

File: src/app/categories/categories.service.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import { Observable } from 'rxjs/Observable';
+import { AngularFirestoreCollection, AngularFirestore } from 'angularfire2/firestore';
+
+import { Category } from './category.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CategoriesService {
+  collection: AngularFirestoreCollection<Category>;
```
Comment (AndreyLukashenko) [inline src/app/categories/categories.service.ts]: Make this property private, if you're not going to use it somewhere else

File: src/app/categories/category.model.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export interface Category {
+  name: string;
+}
```
Comment (AndreyLukashenko) [inline src/app/categories/category.model.ts]: Hmm 🤔, it's interesting you declare category interface with only one property, but in the template you use `recipies_amount ` and `image ` properties.

File: src/app/categories/categories.component.html (hunk 1 above)
Comment (dgutoryan) [reply src/app/categories/categories.component.html]: Done

File: src/app/categories/categories.component.html (hunk 2 above)
Comment (dgutoryan) [reply src/app/categories/categories.component.html]: Done. Firebase returned `string` instead `number`. Fix type in firebase.

File: src/app/categories/categories.service.ts (hunk 3 above)
Comment (dgutoryan) [reply src/app/categories/categories.service.ts]: Done

File: src/app/categories/category.model.ts (hunk 4 above)
Comment (dgutoryan) [reply src/app/categories/category.model.ts]: Done

**Pull request merged**